### PR TITLE
NMS-12445: Integrate Graph Service with the opennms-integration-api (OIA)

### PR DESCRIPTION
**Pull request by mvrueden**
Here we integrate the new Graph Service API with the OpenNMS Integration API.

JIRA: https://issues.opennms.org/browse/NMS-12445

We may need to do a new OIA release (v.0.3.1) in order to get this to work properly.

### CODE REVIEW

**Comment by j-white** [comment]
Is it possible to have links between namespaces/layers and use the "Navigate To" functionality?

I tried [this](https://github.com/j-white/sample-graph-plugin/blob/master/src/main/java/org/opennms/plugins/graph/MyGraphContainerProvider.java#L83), but I guess I'm missing something.

**Comment by mvrueden** [comment]
The `Navigate To` functionality is not implemented in the `LegacyMetaTopologyProvider` yet. I can look into it.

File: features/graph/service/src/main/java/org/opennms/netmgt/graph/service/DefaultGraphService.java [hunk 1]
```diff
@@ -42,10 +45,25 @@
...
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceRegistration;
+
+import com.google.common.collect.Maps;
 
 public class DefaultGraphService implements GraphService {
 
     private List<GraphContainerProvider> graphContainerProviders = new CopyOnWriteArrayList<>();
+    private final BundleContext bundleContext;
+    private final Map<GraphContainerProvider, ServiceRegistration<GraphContainerProviderRegistration>> serviceRegistrationMap = Maps.newHashMap();
```
[the remark this answers is not in the document]
Comment (mvrueden) [reply features/graph/service/src/main/java/org/opennms/netmgt/graph/service/DefaultGraphService.java]: Yes, we should use a thread-safe map here.

**Comment by mvrueden** [comment]
I addressed the `Navigate To` issue. For now, there is no `Breadcrumb Strategy` available. We may want to add that at a later point.

I fixed the NPE when a topology provider is no longer available (e.g. bundle is stopped). The issue here was, that the cache was not invalidated when the MetaTopologyProvider is unregistered. 

I was not able to reproduce the other NPE in the GraphPainter.
Maybe it was related to the TopologyProvider not available anymore. 
However I noticed one other weird issue, that in some cases neither the icons nor the edges were shown. The reason for this is probably the order in which the services were registered. I altered them slightly and the side-effects seem to be fixed.

Please try again and let me know if everything is now working as suspected.

**Pull request merged**